apps/vscode/src/providers/background.ts

```diff
@@ -162,6 +163,10 @@ async function setEditorHighlightDecorations(
       blockRanges.push(vscRange(block.range));
     }
 
+    // expose cell language for use in keybindings, etc
+    const language = mainLanguage(tokens);
+    vscode.commands.executeCommand('setContext', 'quartoLangId', language?.ids[0]);
```

Comment:
I'd consider using a non-abbreviated and very descriptive name

```
quarto.document.mainLanguageId
quartoDocumentMainLanguageId
```

See `"quarto.assistView.isEnabled"` as a nice example

It's a little tricky because we have a mix of styles right now for context key naming. Sometimes we do `quarto.` sometimes we dont https://github.com/search?q=repo%3Aquarto-dev%2Fquarto%20setContext&type=code

I tend to like using the `.` as a way to namespace it to quarto

Reply:
Why not just `quarto.document.languageId` vs (eventually) something like `quarto.cell.languageId`?

`document` vs `cell` is an evocative distinction. `main` wants to contrast with `secondary`, and that's not what you're setting up to do in the future.

Reply:
Works great for me!